datadog/fwprovider/resource_datadog_shared_dashboard.go

```diff
@@ -0,0 +1,431 @@
+package fwprovider
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV1"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	frameworkPath "github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &sharedDashboardResource{}
+	_ resource.ResourceWithImportState = &sharedDashboardResource{}
+)
+
+type sharedDashboardResource struct {
+	Api  *datadogV1.DashboardsApi
+	Auth context.Context
+}
+
+type sharedDashboardModel struct {
+	ID                          types.String                   `tfsdk:"id"`
+	DashboardId                 types.String                   `tfsdk:"dashboard_id"`
+	DashboardType               types.String                   `tfsdk:"dashboard_type"`
+	GlobalTimeSelectableEnabled types.Bool                     `tfsdk:"global_time_selectable_enabled"`
+	PublicUrl                   types.String                   `tfsdk:"public_url"`
+	ShareType                   types.String                   `tfsdk:"share_type"`
+	Token                       types.String                   `tfsdk:"token"`
+	ShareList                   types.List                     `tfsdk:"share_list"`
```

Comment:
Hey! This field will be marked deprecated soon sorry. We're in the process of updating the API contract for this API.